packages/cubejs-clickhouse-driver/driver/ClickHouseDriver.js

```diff
@@ -11,6 +11,8 @@ class ClickHouseDriver extends BaseDriver {
       host: process.env.CUBEJS_DB_HOST,
       port: process.env.CUBEJS_DB_PORT,
       auth: process.env.CUBEJS_DB_USER || process.env.CUBEJS_DB_PASS ? `${process.env.CUBEJS_DB_USER}:${process.env.CUBEJS_DB_PASS}` : '',
+      protocol: process.env.CUBEJS_DB_SSL ? 'https:' : 'http:',
+      readonly: Boolean(process.env.CUBEJS_DB_CLICKHOUSE_READONLY),
```

Comment:
I dont think that We should introduce a new env variable called `CUBEJS_DB_CLICKHOUSE_READONLY`, because it's not universal for all drivers.

Reply:
Let me explain! In ClickHouse, there's this [readonly](https://clickhouse.tech/docs/en/operations/settings/permissions-for-queries/#settings_readonly) option which defined the restrictions on the user's permissions:

> 0 — All queries are allowed.
> 1 — Only read data queries are allowed.
> 2 — Read data and change settings queries are allowed.

So, it makes sense to introduce a new env variable called `CUBEJS_DB_CLICKHOUSE_READONLY` to reflect this feature of ClickHouse. As you can see in the [docs](https://github.com/apla/node-clickhouse#options) for the library we use to connect to ClickHouse, they also introduce this kind of config option.